presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,392 @@
+            throw new RuntimeException("Could not create empty file");
+        }
+        return temporaryDirectory;
+    }
+
+    /**
+     * Generates a temporary directory and inserts data in every partition of the bucketed table, including an empty file in the first partition
+     *
+     * @param queryRunner a {@link QueryRunner} with the desired configuration properties
+     * @param tableName a {@link String} containing the desired table name
+     * @return a {@link File} pointing to the newly created temporary directory
+     */
+    private File generateBucketedMetadata(DistributedQueryRunner queryRunner, String tableName, boolean replace) throws Exception
+    {
+        URL url = TestHiveSkipEmptyFiles.class.getClassLoader().getResource(".");
+        if (url == null) {
+            throw new RuntimeException("Could not obtain resource URL");
+        }
+        File temporaryDirectory = new File(url.getPath(), tableName);
+        boolean created = temporaryDirectory.mkdirs();
+        if (!created) {
+            throw new RuntimeException("Could not create resource directory: " + temporaryDirectory.getPath());
+        }
+        @Language("SQL") String createQuery = format("CREATE TABLE %s.\"%s\".\"%s\" (id %s, field %s) WITH (external_location = '%s'," +
+                        "format = 'Parquet',partitioned_by = ARRAY['field']," +
+                        "  bucketed_by = ARRAY['id']," +
+                        "  bucket_count = 3)",
+                CATALOG, SCHEMA, tableName, IntegerType.INTEGER, VarcharType.VARCHAR, getResourceUrl(tableName));
+        queryRunner.execute(createQuery);
+        String partitionDirectoryPath = temporaryDirectory.getPath() + "/field=field1";
+        String name = "field";
+        @Language("SQL") String insertQuery;
+        for (int i = 1; i <= 5; i++) {
+            insertQuery = format("INSERT INTO %s.\"%s\".\"%s\" VALUES (%s,'%s')",
+                    CATALOG, SCHEMA, tableName, i, name + i);
+            queryRunner.execute(insertQuery);
+            if (i == 1) {
+                File secondParquetFile = new File(partitionDirectoryPath, randomUUID().toString());
+                if (!secondParquetFile.createNewFile()) {
+                    throw new RuntimeException("Could not create empty file");
+                }
+            }
+        }
+        if (replace) {
+            File partitionDirectory = new File(partitionDirectoryPath);
+            deleteMetadata(partitionDirectory, true);
+        }
+        return temporaryDirectory;
+    }
+
+    /**
+     * Deletes the given directory and all of its contents recursively
+     * Does not follow symbolic links
+     *
+     * @param temporaryDirectory a {@link File} pointing to the directory to delete
+     * @param hasPattern a {@code true} if it is necessary to delete only one directory file, {@code false} otherwise
+     */
+    private static void deleteMetadata(File temporaryDirectory, boolean hasPattern)
```

Comment:
You've done far more work here than you needed, I'm afraid.

Reply:
Fixed